### PR TITLE
A few battle/partner funcs

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/battle/partner/watt.c [hunk 1]
```diff
@@ -173,7 +265,45 @@ ApiStatus func_80238C08_7046F8(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "battle/partner/watt", func_80238C84_704774);
+ApiStatus func_80238C84_704774(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    EffectInstance* temp_a0;
```
Comment (savaughn) [inline src/battle/partner/watt.c:270]: looks like this might be unused

File: src/battle/partner/lakilester.c [hunk 2]
```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
...
+            D_8023D2A0 = dpadY;
+            D_8023D2A8 = D_8023D2A4;
+            temp_f10 = D_8023D2A0 - (temp_f2 / 3.0f);
+            D_8023D2A4 = temp_f8;
+            if (temp_f10 > 0) {
+                D_8023D2AC = temp_f10;
+                if (partner->state.speed != 0.0f) {
+                    partner->state.speed += (D_8023D2AC / 8.5) / 3.0;
+                } else {
+                    partner->state.speed += (D_8023D2AC / 8.5) * 0.5;
```
Comment (HailSanta) [inline src/battle/partner/lakilester.c:173]: / 2.0 to match alternate case

File: src/battle/partner/lakilester.c [hunk 3]
```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
...
+            } else {
+                D_8023D2AC = temp_f10;
+            }
+            if (!(dpadY < 31)) {
+                D_8023D2B4 = dpadX;
+            }
+            if (partnerState->speed > 6.7) {
+                partnerState->speed = 6.7f;
+            }
+            partnerState->speed = partnerState->speed - (f32) ((2.0 * (f64) (0x28 - dpadY)) / 40.0);
```
Comment (HailSanta) [inline src/battle/partner/lakilester.c:185]: casts needed?

File: src/battle/partner/lakilester.c [hunk 4]
```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
...
+                hud_element_free(id);
+            }
+            btl_set_popup_duration(0);
+            sfx_stop_sound(0x312);
+            return ApiStatus_DONE2;
+    }
+
+    get_screen_coords(gCurrentCameraID,
+                      partnerState->goalPos.x, partnerState->goalPos.y, partnerState->goalPos.z,
+                      &screenX, &screenY, &sp28);
```
Comment (HailSanta) [inline src/battle/partner/lakilester.c:245]: sp28 = screenZ

File: src/battle/partner/lakilester.c [hunk 5]
```diff
@@ -19,7 +44,244 @@ ApiStatus func_80238000_70BD10(Evt* script, s32 isInitialCall) {
...
+    if (temp_v1_2 < 3 && temp_v1_2 > 0) {
+        D_8023D2C0 += 20;
+        if (D_8023D2C0 > 50) {
+            D_8023D2C0 = 50;
+        }
+        hud_element_set_render_pos(D_8023D2BC, D_8023D2C0, D_8023D2C4);
+    }
+
+    temp_a0_2 = script->functionTemp[0];
+    if (temp_a0_2 < 3) {
```
Comment (HailSanta) [inline src/battle/partner/lakilester.c:266]: can these be combined, as a few lines above?

**Review by HailSanta — APPROVED** [review]

File: src/battle/partner/lakilester.c (hunk 3 above)
Comment (ethteck) [reply src/battle/partner/lakilester.c:185]: the first one is, but I'll remove the second

File: src/battle/partner/lakilester.c (hunk 5 above)
Comment (ethteck) [reply src/battle/partner/lakilester.c:266]: this section is part of why the function doesn't match - I'd prefer to leave it as-is for now until we figure out how to match it

Comment (ethteck) [reply src/battle/partner/lakilester.c:266]: combining didn't help when I was trying to match it earlier

**Pull request merged**